plugins/modules/ndo_ipsla_track_list.py

```diff
@@ -0,0 +1,468 @@
+                )
+            )
+
+    ops = []
+    match = None
+
+    # The object dictionary is used as a cache store for schema & template data.
+    # This is done to limit the amount of API calls when UUID is not specified for member scope references.
+    obj_cache = {}
+
+    mso_template = MSOTemplate(mso, "tenant", template)
+    mso_template.validate_template("tenantPolicy")
+    obj_cache["template-tenant-{0}".format(template)] = mso_template
+
+    object_description = "IPSLA Track List"
+    path = "/tenantPolicyTemplate/template/ipslaTrackLists"
+
+    existing_ipsla_track_lists = mso_template.template.get("tenantPolicyTemplate", {}).get("template", {}).get("ipslaTrackLists", [])
+    if ipsla_track_list or ipsla_track_list_uuid:
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            existing_ipsla_track_lists,
+            [(KVPair("uuid", ipsla_track_list_uuid) if ipsla_track_list_uuid else KVPair("name", ipsla_track_list))],
+        )
+        if match:
+            mso.existing = mso.previous = copy.deepcopy(match.details)
+    else:
+        mso.existing = mso.previous = existing_ipsla_track_lists
+
+    if state == "present":
+        if ipsla_track_list_uuid and not mso.existing:
+            mso.fail_json(msg="{0} with the UUID: '{1}' not found".format(object_description, ipsla_track_list_uuid))
+
+        mso_values = {
+            "name": ipsla_track_list,
+            "description": description,
+            "type": list_type,
+            "{0}Up".format(list_type): thresholds["up"],
+            "{0}Down".format(list_type): thresholds["down"],
+        }
+        if members is not None:
+            mso_values["trackListMembers"] = format_track_list_members(mso, mso_template, members, obj_cache)
+        if match:
+            append_update_ops_data(ops, match.details, "{0}/{1}".format(path, match.index), mso_values)
+            mso.sanitize(match.details, collate=True)
+        else:
+            mso.sanitize(mso_values)
+            ops.append(dict(op="add", path="{0}/-".format(path), value=mso.sent))
+
+    elif state == "absent":
+        if match:
+            ops.append(dict(op="remove", path="{0}/{1}".format(path, match.index)))
+
+    if not module.check_mode and ops:
+        response = mso.request(mso_template.template_path, method="PATCH", data=ops)
+        ipsla_track_lists = response.get("tenantPolicyTemplate", {}).get("template", {}).get("ipslaTrackLists", [])
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            ipsla_track_lists,
+            [(KVPair("uuid", ipsla_track_list_uuid) if ipsla_track_list_uuid else KVPair("name", ipsla_track_list))],
+        )
+        if match:
+            mso.existing = match.details
+        else:
+            mso.existing = {}
+    elif module.check_mode and state != "query":
+        mso.existing = mso.proposed if state == "present" else {}
```

Comment:
Insert the `ipslaMonitoringRef` and `scope` object name in the `current, previous` output portion when we do the `create, update, query, delete`.